src/main/java/org/intellij/plugin/mdx/js/MdxTagNameSynchronizer.java

```diff
@@ -0,0 +1,424 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.intellij.plugin.mdx.js;
+
+import com.intellij.application.options.editor.WebEditorOptions;
+import com.intellij.codeInsight.lookup.LookupManager;
+import com.intellij.codeInsight.lookup.impl.LookupImpl;
+import com.intellij.codeInspection.htmlInspections.RenameTagBeginOrEndIntentionAction;
+import com.intellij.ide.plugins.DynamicPluginListener;
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.lang.Language;
+import com.intellij.lang.html.HTMLLanguage;
+import com.intellij.lang.injection.InjectedLanguageManager;
+import com.intellij.lang.xhtml.XHTMLLanguage;
+import com.intellij.lang.xml.XMLLanguage;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.command.CommandEvent;
+import com.intellij.openapi.command.CommandListener;
+import com.intellij.openapi.command.undo.UndoManager;
+import com.intellij.openapi.diagnostic.Attachment;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.editor.*;
+import com.intellij.openapi.editor.event.DocumentEvent;
+import com.intellij.openapi.editor.event.DocumentListener;
+import com.intellij.openapi.editor.event.EditorFactoryEvent;
+import com.intellij.openapi.editor.event.EditorFactoryListener;
+import com.intellij.openapi.editor.impl.EditorImpl;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Couple;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.pom.core.impl.PomModelImpl;
+import com.intellij.psi.*;
+import com.intellij.psi.impl.PsiDocumentManagerBase;
+import com.intellij.psi.impl.source.tree.TreeUtil;
+import com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil;
+import com.intellij.psi.templateLanguages.OuterLanguageElement;
+import com.intellij.psi.templateLanguages.TemplateLanguage;
+import com.intellij.psi.templateLanguages.TemplateLanguageUtil;
+import com.intellij.psi.xml.XmlTokenType;
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.xml.XmlExtension;
+import com.intellij.xml.util.HtmlUtil;
+import com.intellij.xml.util.XmlUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * It is a copy-paste, only SUPPORTED_LANGUAGES is changed
+ *
+ * @see com.intellij.codeInsight.editorActions.XmlTagNameSynchronizer
+ */
+
+public final class MdxTagNameSynchronizer implements EditorFactoryListener {
+    private static final Key<Boolean> SKIP_COMMAND = (Key<Boolean>) Key.findKeyByName("tag.name.synchronizer.skip.command");
```

Comment:
we need to make sure that XmlTagSynchronizer is already loaded + also we need to handle the exception if the key isn't found